### PR TITLE
Client profiles for TSH

### LINKED ISSUE #578: Add login profiles to TSH
`tsh` takes a lot of parameters to connect to machine. It would make Teleport much easier to use if `tsh` could remember the last used cluster and/or connection parameters.

Here's what users do today using bash tools like aliases:

```
alias telelocal="export TELEPORT_LOGIN=admin; export TELEPORT_PROXY=localhost:3024,33009; export TELEPORT_AUTH=google; export TELEPORT_SITE=dev.local; tsh ssh --local -L 8080:localhost:8080 -L 5000:localhost:5000 root@gravitational.io/k8s-role=master"
```

would be nice if `tsh` could support this natively.

### PR DESCRIPTION
**Pull request by kontsevoy**
## Description

This PR adds  support for client profiles for `tsh`. A profile is a collection of commonly used CLI arguments like `--proxy` or `-L` stored in a simple YAML file in `~/.tsh` directory.

Profiles are created automatically, one per proxy server. The profile name is `~/.tsh/proxyhost.yaml`. Additionally, the last successful login always creates `~/.tsh/profile` symlink pointing to the profile, which means `tsh` remembers the last successful setting.

Closes #578 
#### Port sequence in --proxy argument

Plenty of people run proxy HTTPS on port 443. They need to tell TSH to use that port. Currently when you say `tsh --proxy=host:port,port2` the "port" part is interprented as SSH proxy port, not HTTPS port (that would be "port2"). This PR swaps them, so they can do "proxyhost:443" now. 
#### Error messages

If you use the wrong port number in `tsh` arguments, you see error messages that are hard to interpret (like "invalid subsystem"). This PR adds friendlier error messages, suggesting that users are connecting to the wrong service (port).

### CODE REVIEW

**Comment by klizhentas** [comment]
retest this please

**Comment by kontsevoy** [comment]
retest this please

**Comment by kontsevoy** [comment]
retest this please

File: lib/client/profile.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+	// SiteName is equivalient to --cluster argument
+	SiteName string `yaml:"cluster,omitempty"`
+
+	//
+	// other stuff
+	//
+	ForwardedPorts []string `yaml:"forward_ports,omitempty"`
+}
+
+// fullProfilePath returns the full path to the user profile directory.
```
Comment (klizhentas) [inline lib/client/profile.go]: FullProfilePath

File: tool/tsh/main.go [hunk 2]
```diff
@@ -311,6 +318,9 @@ func onSCP(cf *CLIConf) {
 		} else {
 			utils.FatalError(err)
 		}
+	} else {
+		// successful session? update the profile then:
+		tc.SaveProfile("")
```
Comment (klizhentas) [inline tool/tsh/main.go]: what's that for?

File: tool/tsh/main.go [hunk 3]
```diff
@@ -383,24 +393,34 @@ func makeClient(cf *CLIConf) (tc *client.TeleportClient, err error) {
...
-		ConnectorID:        cf.ExternalAuth,
-		SiteName:           cf.SiteName,
-		Interactive:        cf.Interactive,
-	}
+	// 1: start with the defaults
+	c := client.MakeDefaultConfig()
+
+	// 2: override with `./tsh` profiles
+	if err = c.LoadProfile(""); err != nil {
+		fmt.Printf("WARNING: Failed loading tsh profile.\n%v\n", err)
```
Comment (klizhentas) [inline tool/tsh/main.go]: why is that warning?

**Review by klizhentas — APPROVED** [review]

File: tool/tsh/main.go (hunk 2 above)
Comment (kontsevoy) [inline tool/tsh/main.go]: this saves the most recent profile.

File: tool/tsh/main.go (hunk 3 above)
Comment (kontsevoy) [inline tool/tsh/main.go]: because failing doesn't make sense (user may have supplied enough CLI flags to proceed) and we don't want to hide the error.

**Comment by kontsevoy** [comment]
updated.

**Review by klizhentas — APPROVED** [review]

**Pull request merged**